eth/downloader/downloader.go

```diff
@@ -1520,7 +1538,7 @@ func (d *Downloader) processHeaders(origin uint64, pivot uint64, td *big.Int) er
 						n, newCanonical, lowestCanonicalNumber, err = d.blockchain.InsertHeaderChainStaged(chunk, frequency)
 						if newCanonical {
 							// Need to unwind further stages
-							if err1 := UnwindAllStages(d.stateDB, lowestCanonicalNumber); err1 != nil {
+							if err1 := UnwindAllStages(d.stateDB, lowestCanonicalNumber, d.quitCh); err1 != nil {
```

Comment:
This is not a long running function, and it needs to execute atomically, giving it interruptibility is overkill and is not required